internal/controllers/machinedeployment/machinedeployment_status.go

```diff
@@ -40,20 +39,14 @@ import (
 
 func (r *Reconciler) updateStatus(ctx context.Context, s *scope) (retErr error) {
 	// Get all Machines controlled by this MachineDeployment.
-	var machines, machinesToBeRemediated, unhealthyMachines collections.Machines
+	var machinesToBeRemediated, unhealthyMachines collections.Machines
 	var getMachinesSucceeded bool
-	if selectorMap, err := metav1.LabelSelectorAsMap(&s.machineDeployment.Spec.Selector); err == nil {
-		machineList := &clusterv1.MachineList{}
-		if err := r.Client.List(ctx, machineList, client.InNamespace(s.machineDeployment.Namespace), client.MatchingLabels(selectorMap)); err != nil {
-			retErr = errors.Wrap(err, "failed to list machines")
-		} else {
-			getMachinesSucceeded = true
-			machines = collections.FromMachineList(machineList)
-			machinesToBeRemediated = machines.Filter(collections.IsUnhealthyAndOwnerRemediated)
-			unhealthyMachines = machines.Filter(collections.IsUnhealthy)
-		}
+	if s.machines != nil {
+		getMachinesSucceeded = true
+		machinesToBeRemediated = s.machines.Filter(collections.IsUnhealthyAndOwnerRemediated)
+		unhealthyMachines = s.machines.Filter(collections.IsUnhealthy)
 	} else {
-		retErr = errors.Wrap(err, "failed to convert label selector to a map")
+		retErr = errors.Errorf("failed to convert label selector to a map")
```

Comment:
I think we don't know if that error was the reason why we couldn't get Machines. But see my other comment in machinedeployment_controller.go.

If we follow that comment this here becomes a lot easier (we can just always assume we were able to get Machines)

(Note to myself: in any case re-review this change)